### PR TITLE
Feat(auth-gaurd)

### PR DESCRIPTION
**Pull request by sijav**
Add auth gaurd, fix graphql, add error boundaries and login dummy page

### CODE REVIEW

File: src/shared/error-boundary/ErrorBoundary.tsx [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+}
+
+export class ErrorBoundary extends React.Component<Props, State> {
+  constructor(props: Props) {
+    super(props);
+    this.state = { error: null };
+  }
+
+  static getDerivedStateFromError(error: Error) {
+    return { error: error };
```
Comment (c0m1t) [inline src/shared/error-boundary/ErrorBoundary.tsx]: { error }

File: src/pages/loading/FullPageSpinner.tsx [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+import { Styles } from 'src/shared/types/Styles';
+
+interface OwnProps {}
+
+type Props = FCProps<OwnProps> & StyleProps;
+
+export function FullPageSpinner(props: Props) {
+  const classes = useStyles(props);
+  return (
+    <div className={classes.root}>
```
Comment (c0m1t) [inline src/pages/loading/FullPageSpinner.tsx]: styled component?

File: src/pages/auth/Login.tsx [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+import { FCProps } from 'src/shared/types/FCProps';
+import { Styles } from 'src/shared/types/Styles';
+
+interface OwnProps {}
+
+type Props = FCProps<OwnProps> & StyleProps;
+
+export function Login(/* props: Props */) {
+  // const classes = useStyles();
+  return <div>{i18n._('Login')}</div>;
```
Comment (c0m1t) [inline src/pages/auth/Login.tsx]: extract and translate?

File: src/pages/loading/FullPageSpinner.tsx (hunk 2 above)
Comment (smmoosavi) [reply src/pages/loading/FullPageSpinner.tsx]: more than one element is in this component, so useStyle is better

**Pull request merged**